packages/multichain-account-service/src/providers/SnapAccountProvider.ts

```diff
@@ -222,14 +286,32 @@ export abstract class SnapAccountProvider extends BaseBip44AccountProvider {
     );
   }
 
+  async createAccounts(options: {
+    entropySource: EntropySourceId;
+    groupIndex: number;
+  }): Promise<Bip44Account<KeyringAccount>[]> {
+    await this.ensureSnapPlatformIsReady();
+
+    return await this.runCreateAccounts(options);
+  }
+
+  async discoverAccounts(options: {
+    entropySource: EntropySourceId;
+    groupIndex: number;
+  }): Promise<Bip44Account<KeyringAccount>[]> {
+    await this.ensureSnapPlatformIsReady();
+
+    return await this.runDiscoverAccounts(options);
+  }
+
   abstract isAccountCompatible(account: Bip44Account<InternalAccount>): boolean;
 
-  abstract createAccounts(options: {
+  abstract runCreateAccounts(options: {
```

Comment:
Now providers have to implements those `run*` methods, so we can automatically call `this.ensureSnapPlatformIsReady` for any of those actions.